nil/services/synccommittee/prover/tracer/state_db.go

```
@@ -378,12 +386,29 @@ func (tsdb *TracerStateDB) GetTransactionFlags() types.TransactionFlags {
 	panic("not implemented")
 }
 
-func (tsdb *TracerStateDB) GetTokens(types.Address) map[types.TokenId]types.Value {
-	panic("not implemented")
+func (tsdb *TracerStateDB) GetTokens(addr types.Address) map[types.TokenId]types.Value {
+	acc, err := tsdb.mptTracer.GetAccount(addr)
+	check.PanicIfErr(err)
+	if acc == nil {
+		return nil
+	}
+
+	res := make(map[types.TokenId]types.Value)
+	for k, v := range acc.TokenTree.Iterate() {
+		var c types.TokenBalance
+		c.Token = types.TokenId(k)
+		if err := c.Balance.UnmarshalSSZ(v); err != nil {
+			tsdb.logger.Error().Err(err).Msg("failed to unmarshal token balance")
```

If SSZ serialization is somehow broken this code will silently return empty map instead of explicit indicating the problem. Are you sure we want this behavior?

---

Just the same flow as in original [state.go](https://github.com/NilFoundation/nil/blob/9d091d81a06cf437a18318fa39308364650699b2/nil/internal/execution/state.go#L1642). If there will be such issue, we need to copy the behaviour.